components/node-setup/init-node.sh

```diff
@@ -49,16 +49,39 @@ else
   logme "${PRIVATEDIR}/sockets/app.sock removed"
 fi
 
+
 # create a private key
 if [ "$RESETALL" == "true" ]; then
   if ! [ -f "${PRIV_KEY}" ]; then
-   openssl genrsa -out "${PRIV_KEY}" 2048
-   openssl req -new -sha256 -key "${PRIV_KEY}" -out "${SERV_CSR}" -subj "/CN=jss-jwt-server"
-   logme "private key ${PRIV_KEY} and ${SERV_CSR} created"
+
+    if [ "$CLUSTER_REGION" == "" ] || [ "$CLUSTER_NAME" == "" ]; then
+      logme "Env. variables CLUSTER_REGION and CLUSTER_NAME must be set. Terminating..."
+    else
+      cat > "openssl.cnf" << EOF
+[req]
+req_extensions = v3_req
+
+[v3_req]
+subjectAltName= @alt_names
+
+[alt_names]
+URI.1 = TSI:cluster-name:$CLUSTER_NAME
+URI.2 = TSI:cluster-region:$CLUSTER_REGION
+# URI.3 = TSI:datacenter:fra02
```

Comment:
let's add a comment here that says something along the lines of "If you'd like to assert certain claims about this intermediate CA, add additional lines in the format

```
URI.X = TSI:<claim>
```
where X is a sequential number and <claim> is a ":" delimetered key value pair

Reply:
Done